src/sirc/sirc_context.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2021 Val Lorentz <progval+srain@progval.net>
+ *
+ * This file is part of Srain.
+ *
+ * Srain is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "sirc/sirc.h"
+
+struct _SircMessageContext {
+    GDateTime *time; 
+};
+
+SircMessageContext* sirc_message_context_new(GDateTime* time) {
+    SircMessageContext *context;
+
+    if (!time) {
+        time = g_date_time_new_now_local();
+    }
+
+    g_return_val_if_fail(time, NULL);
```

Comment:
This check can be omitted, I think.

Reply:
"This function will always succeed unless GLib is still being used after the year 9999."

fair enough